jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -788,6 +789,26 @@ MethodHandle insertArguments(MethodHandle equivalent, MethodHandle unboxingHandl
 		return result;
 	}
 
+	private static final Class<?> fromFieldDescriptorString(String fieldDescriptor, ClassLoader classLoader) {
+		ArrayList<Class<?>> classList = new ArrayList<Class<?>>();
+		int length = fieldDescriptor.length();
+		if (length == 0) {
+			/*[MSG "K05d3", "invalid descriptor: {0}"]*/
+			throw new IllegalArgumentException(Msg.getString("K05d3", fieldDescriptor)); //$NON-NLS-1$
+		}
+
+		char[] signature = new char[length];
+		fieldDescriptor.getChars(0, length, signature, 0);
+
+		MethodType.parseIntoClass(signature, 0, classList, classLoader, fieldDescriptor);
+
+		if (classList.isEmpty()) {
+			throw new NoClassDefFoundError(fieldDescriptor);
+		}
```

Comment:
since parseIntoClass will only parse one valid descriptor and append that to the list or throw an exception, I don't think this check is even needed.

I'll remove the check